leaflet-treering.js

```diff
@@ -348,19 +348,72 @@ function MeasurementData (dataObject) {
    * @function insertPoint
    */
   MeasurementData.prototype.insertPoint = function(latLng, hasLatewood) {
-    var i = 0;
-    while (this.points[i] != undefined &&
-        this.points[i].latLng.lng < latLng.lng) {
-      i++;
-    }
+    var disList = [];
+
+    // finds point with smallest abs. distance
+    for (i = 0; i < this.points.length; i++) {
+      var distance = Number.MAX_SAFE_INTEGER;
+      if (this.points[i] && this.points[i].latLng) {
+         var currentPoint = this.points[i].latLng;
+         distance = Math.sqrt(Math.pow((latLng.lng - currentPoint.lng), 2) + Math.pow((latLng.lat - currentPoint.lat), 2));
+      }
+      disList.push(distance);
+    };
+
+    var minDistance = Math.min(...disList);
+    i = disList.indexOf(minDistance)
+
     if (this.points[i] == null) {
-      alert('New point must be within existing points.' +
-          'Use the create toolbar to add new points to the series.');
+      alert('New point must be within existing points. Use the create toolbar to add new points to the series.');
       return;
     }
 
+    // define 3 points: i, i - 1, i + 1
+    var iPoint = this.points[i].latLng;
+
+    if (this.points[i - 1].start) {
+      var iMinus = this.points[i].latLng;
+    } else {
+      var iMinus = this.points[i - 1].latLng;
+    };
+
+    if (this.points[i + 1]) { //
+      var iPlus = this.points[i + 1].latLng;
+    } else {
+      var iPlus = L.latLng(0.15, 0.6);
+    };
+
+    // distance: point[i] to point[i + 1]
+    var disIPlus = Math.sqrt(Math.pow((iPlus.lng - iPoint.lng), 2) +
```

Comment:
This is a time where doing an inline function makes sense to me.  

var computeDistance = function(point1, point2) { return Math.sqrt(Math.pow((point1.lng - point2.lng), 2) +
                               Math.pow((point1.lat - point2.lat), 2)); };